packages/permission-controller/src/rpc-methods/revokePermissions.test.ts

```diff
@@ -0,0 +1,93 @@
+import { JsonRpcEngine } from '@metamask/json-rpc-engine';
+import { rpcErrors } from '@metamask/rpc-errors';
+
+import { revokePermissionsHandler } from './revokePermissions';
+
+describe('revokePermissions RPC method', () => {
+  it('returns the values of the object returned by requestPermissionsForOrigin', async () => {
+    const { implementation } = revokePermissionsHandler;
+    const mockRevokePermissionsForOrigin = jest.fn();
+
+    const engine = new JsonRpcEngine();
+    engine.push((req, res, next, end) =>
+      implementation(req as any, res as any, next, end, {
+        revokePermissionsForOrigin: mockRevokePermissionsForOrigin,
+      }),
+    );
+
+    const response: any = await engine.handle({
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'wallet_revokePermissions',
+      params: { snap_dialog: { caveats: [{ type: 'foo', value: 'bar' }] } },
```

Comment:
does adding the caveats here only remove that particular caveat?

Reply:
Not for this implementation no, but the idea is that we can add that down the line without breaking the API.

So we allow passing in caveats now, but they are effectively ignored until we know how to handle them.

Reply:
the previous PR to extension used `permissionController.updateCaveat`, is there anything wrong with that approach? https://github.com/MetaMask/metamask-extension/pull/21216/files#diff-6fbff2cfe97ac01b77296ef2122c7e0a5b3ff6a84b584b4d1a87482f35eea3d6R3779

Reply:
Yeah, it is a flawed approach. Caveat values can be arbitrary JSON and are not standardized: https://github.com/MetaMask/core/blob/main/packages/permission-controller/src/Caveat.ts#L31C19-L31C23

Therefore it is not safe to assume we can modify them in this way.

We need to standardize the caveat format (effectively allowing for caveat merging, easy removal etc) before we implement this. That would be the next iteration of this API, but we have yet to discuss making that change in the PermissionController.